include/exadg/grid/grid.h

```diff
@@ -73,6 +74,124 @@ class Grid
     mapping = std::make_shared<dealii::MappingQGeneric<dim>>(data.mapping_degree);
   }
 
+  void
+  create_and_refine_triangulation(
+    GridData const &                                          data,
+    std::vector<int> const &                                  vector_local_refinements,
+    std::function<void(dealii::Triangulation<dim> &)> const & create_coarse_triangulation)
+  {
+    if(data.triangulation_type == TriangulationType::Serial or
+       data.triangulation_type == TriangulationType::Distributed)
+    {
+      create_coarse_triangulation(*triangulation);
+
+      if(vector_local_refinements.size() > 0)
+        refine_local(*triangulation, vector_local_refinements);
+
+      triangulation->refine_global(data.n_refine_global);
+    }
+    else if(data.triangulation_type == TriangulationType::FullyDistributed)
+    {
+      auto const serial_grid_generator = [&](dealii::Triangulation<dim, dim> & tria_serial) {
+        create_coarse_triangulation(tria_serial);
+
+        if(vector_local_refinements.size() > 0)
+          refine_local(tria_serial, vector_local_refinements);
+
+        tria_serial.refine_global(data.n_refine_global);
+      };
+
+      auto const serial_grid_partitioner = [&](dealii::Triangulation<dim, dim> & tria_serial,
+                                               MPI_Comm const                    comm,
+                                               unsigned int const                group_size) {
+        (void)group_size;
+        if(data.partitioning_type == PartitioningType::Metis)
+        {
+          dealii::GridTools::partition_triangulation(dealii::Utilities::MPI::n_mpi_processes(comm),
+                                                     tria_serial);
+        }
+        else if(data.partitioning_type == PartitioningType::p4est)
+        {
+          dealii::GridTools::partition_triangulation_zorder(
+            dealii::Utilities::MPI::n_mpi_processes(comm), tria_serial);
+        }
+        else
+        {
+          AssertThrow(false, dealii::ExcNotImplemented());
+        }
+      };
+
+      unsigned int const group_size = 1;
```

Comment:
I guess we could turn this into an input argument.

Reply:
Yes. The question is whether we should do it now or later when we need it. In the past, I only introduced parameters when we really want to make use of them.

Reply:
let's do this later